packages/cli/src/dev.ts

```diff
@@ -266,24 +274,38 @@ unregister();
               entrypoint.source
             );
             entrypointPromises.push(
-              hasDefaultExportPromise.then((hasDefaultExport) =>
-                fs.writeFile(
-                  path.join(
-                    entrypoint.package.directory,
-                    getExportsImportUnwrappingDefaultOutputPath(entrypoint)
+              hasDefaultExportPromise.then((hasDefaultExport) => {
+                const filepath = path.join(
+                  entrypoint.package.directory,
+                  getExportsImportUnwrappingDefaultOutputPath(entrypoint)
+                );
+                const importPath = `./${getBaseDistFilename(
+                  entrypoint,
+                  "cjs"
+                )}`;
+                return Promise.all([
+                  fs.writeFile(
+                    filepath,
+                    mjsTemplate(
+                      // the * won't really do anything right now
+                      // since cjs-module-lexer won't find anything
+                      // but that could be fixed by adding fake things
+                      // to the .cjs.js file that look like exports to cjs-module-lexer
+                      // but don't actually add the exports at runtime like esbuild does
+                      // (it would require re-running dev when adding new named exports)
```

Comment:
I don't understand what's happening here but it sounds like it does nothing *today* so that's... fine? 😅

Reply:
correct (this bit is also not new to this PR, it's just moved around)

The comment is generally about making this work in `preconstruct dev`:

```
// somewhere.mjs
import { something } from 'my-pkg'
```

It currently doesn't (and didn't before any of the `importConditionDefaultExport ` stuff).